### ISSUE TITLE
dApps - Zapper.fi transactions can be sign approved but cannot be completed

### ISSUE THREAD

**Issue by virginiabalducci** [status-im/status-desktop#16004]
# Bug Report

## Description

When the Status wallet account is connected to Zapper.xyz via Wallet Connect, and user wants to perform a transaction (Bridge or Swap), the transaction can be sign approved but cannot be completed. 

## Steps to reproduce

1. Log in with an account who has enough funds to try transactions
2. Go to https://zapper.xyz/ and connect the Status wallet via Wallet Connect
3. Start a swap or bridge transaction
4. Sign the "sign request" pop up on Status app
5. On Zapper, see that nothing happens. The transaction does not complete.

#### Expected behavior
After the sign request on Status app,  user should be able to confirm and perform the transaction 

#### Actual behavior
After the sign request on Status app has been approved, nothing happens on the dApp's side, as if the Sign authorization request from Status app never reached the Zapper server.
Also worth mentioning that while I was trying to do a transaction, the activity list history was deleted. This account has plenty of activity, including a successful transaction on Aave Finance (deposit and withdraw) 

https://github.com/user-attachments/assets/a33f2346-3047-410e-aaac-fc388cdda2e0

https://github.com/user-attachments/assets/9204068e-69ec-42b9-ac5b-607793090e33

Bridge test
https://github.com/user-attachments/assets/86264032-b252-421e-acad-4e5f2a7ea8b9

Swap test
https://drive.google.com/file/d/12UFoaMnqJixfOAbz_XO6axRXd1oZdPIz/view?usp=sharing
https://drive.google.com/file/d/1MQctij9Ldp-55siev4irlvYHbV2CagPJ/view?usp=sharing

Wallet address: 0x32e12fd7c2c7984dd6de153691d901445fc8aa07

Console error 
[console log zapper.txt](https://github.com/user-attachments/files/16515096/console.log.zapper.txt)
```
ERR 2024-08-06 16:33:55.086-03:00 qt error                                   topics="qt" tid=536262 text=">Error finding event for topic 01f4dfa567545130f8b4659b9efe2724c09bf0e2d09aa7a9480da3a166243ec9 id 1722972827851972" file=qrc:/app/AppLayouts/Wallet/services/dapps/DappsConnectorSDK.qml:296 category=qml
DBG 2024-08-06 16:33:55.483-03:00 qt message                                 topics="qt" tid=536262 text="WC WalletConnectSDK.onAcceptSessionRequestResponse; topic: 01f4dfa567545130f8b4659b9efe2724c09bf0e2d09aa7a9480da3a166243ec9, id: 1722972827851972 error: " file=qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:398 category=qml
DBG 2024-08-06 16:33:55.484-03:00 qt message                                 topics="qt" tid=536262 text="WC WalletConnectSDK.wcCall.getActiveSessions;" file=qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:127 category=qml
```

### Additional Information

- Status desktop version: 2.30 rc-4
- Operating System: Mac OS Sonoma

**Comment by anastasiyaig** [comment]
i guess it is 2.31 as dapps are not for 2.30 anymore

**Comment by saledjenic** [comment]
I've just tested this and it worked fine on my side:
- approve tx: https://optimistic.etherscan.io/tx/0x7870ef9cc41876e6f09dcc4b02783b1d230b3467ac95b753ab33ec335e9d4c45
- transform erc20 tx: https://optimistic.etherscan.io/tx/0x279ca279c00eb76970846a978de04b3404a80c0025e6b41a8bdc6aea0e120966

@virginiabalducci could you try it again in the latest master?

**Comment by virginiabalducci** [comment]
I was able to do a swap on [Zapper](https://zapper.xyz/) ✅ 
Arbitrum tx hash
https://arbiscan.io/tx/0x9b0b53f2979a0a92e20cf930170d1536ae88871ae3ba0a237c2314659e236ecf
0x9b0b53f2979a0a92e20cf930170d1536ae88871ae3ba0a237c2314659e236ecf

Bridge L2 to L2 (Arbitrum to Optimism): ✅ 
 0x131ec7ee0f84fa36ef46651687bef0b387da99468ad62c5f03fa2f30ae6a5811
https://optimistic.etherscan.io/tx/0x131ec7ee0f84fa36ef46651687bef0b387da99468ad62c5f03fa2f30ae6a5811

I can't bridge L1 to L2, but I'm getting a block from the dApp's page itself and not from the desktop app: ⚠️ 

https://github.com/user-attachments/assets/3b0e4942-5640-4eed-8140-f30923c0d821
[app_20240923_151252.log](https://github.com/user-attachments/files/17103902/app_20240923_151252.log)
[app_20240923_160359.log](https://github.com/user-attachments/files/17103904/app_20240923_160359.log)
I'm attaching the last two console logs from the desktop app in case this helps. 

I was able to do a Swap transaction and L2 - L2 bridge 

@saledjenic

**Comment by alexjba** [comment]
> I can't bridge L1 to L2, but I'm getting a block from the dApp's page itself and not from the desktop app

@virginiabalducci Does it work on other dApps?

**Comment by virginiabalducci** [comment]
> > I can't bridge L1 to L2, but I'm getting a block from the dApp's page itself and not from the desktop app
> 
> @virginiabalducci Does it work on other dApps?

I tested in other dApps such as https://www.brid.gg/op-mainnet and I'm unable to do this type of bridge from L1 to L2
Is this an issue from the dApp end or from the desktop app? I'm attaching the console log and geth.log for reference

https://github.com/user-attachments/assets/93092448-b920-4d3d-b969-c5d950ed4e8a
[app_20241002_080051.log](https://github.com/user-attachments/files/17229269/app_20241002_080051.log)
[geth.log.zip](https://github.com/user-attachments/files/17229272/geth.log.zip)

cc @saledjenic

**Comment by alexjba** [comment]
> > > I can't bridge L1 to L2, but I'm getting a block from the dApp's page itself and not from the desktop app
> > 
> > 
> > @virginiabalducci Does it work on other dApps?
> 
> I tested in other dApps such as https://www.brid.gg/op-mainnet and I'm unable to do this type of bridge from L1 to L2 Is this an issue from the dApp end or from the desktop app? I'm attaching the console log and geth.log for reference
> 
> https://github.com/user-attachments/assets/93092448-b920-4d3d-b969-c5d950ed4e8a [app_20241002_080051.log](https://github.com/user-attachments/files/17229269/app_20241002_080051.log) [geth.log.zip](https://github.com/user-attachments/files/17229272/geth.log.zip)
> 
> cc @saledjenic

Thank you! Looks like the dApp needs a method we're not implementing. One way to debug this could be to enable all methods and check what the dApp calls for the L1>L2 bridge.

This is what the dApp proposes:
```
        "events": [
          "chainChanged",
          "accountsChanged",
          "message",
          "disconnect",
          "connect"
        ],
        "methods": [
          "eth_accounts",
          "eth_requestAccounts",
          "eth_sendRawTransaction",
          "eth_sign",
          "eth_signTransaction",
          "eth_signTypedData",
          "eth_signTypedData_v3",
          "eth_signTypedData_v4",
          "eth_sendTransaction",
          "personal_sign",
          "wallet_switchEthereumChain",
          "wallet_addEthereumChain",
          "wallet_getPermissions",
          "wallet_requestPermissions",
          "wallet_registerOnboarding",
          "wallet_watchAsset",
          "wallet_scanQRCode",
          "wallet_sendCalls",
          "wallet_getCapabilities",
          "wallet_getCallsStatus",
          "wallet_showCallsStatus"
        ],
```

@saledjenic Do you know if any of these methods are involved in the L1->L2 bridge.

**Comment by saledjenic** [comment]
@virginiabalducci I've just tested a bridge on Zapper, I've sent 1 USDC from Mainnet to Arbitrm using WalletConnect and Status app and here are txs:
- approval: https://etherscan.io/tx/0xe58b4b3bb944581c57386d9e1812423b726b1cd30083a1b668d6cfbb57816d45
- bridge out: https://etherscan.io/tx/0x141d072c32c44ba95a3d7c83cd93df6ff13ef44edf1be43bf6aad2813360b3ce
- bridge in: https://arbiscan.io/tx/0xb271b6e09b003d99f1a77a9331354b3e71fa377568839b63ef3d5bcb2a86ccc8

**Comment by virginiabalducci** [comment]
Hi @saledjenic 

I try to bridge from L1 to L2 Optimism and nothing happens
https://github.com/user-attachments/assets/cb7411b7-e91b-44a1-b1d0-a7a56d51c511

```
DBG 2024-10-04 14:40:04.049Z NewBE_callPrivateRPC                       topics="rpc" tid=51310188 file=core.nim:27 rpc_method=wallet_getWalletConnectActiveSessions
DBG 2024-10-04 14:42:27.902Z NewBE_callPrivateRPC                       topics="rpc" tid=51310188 file=core.nim:27 rpc_method=wakuext_speedupArchivesImport
DBG 2024-10-04 14:43:05.243Z qt message                                 topics="qt" tid=51310188 text="WC unhandled event: \"relayer_disconnect\" connection to the relay server is lost" file=qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:277 category=qml
DBG 2024-10-04 14:43:06.831Z qt message                                 topics="qt" tid=51310188 text="WC unhandled event: \"relayer_connect\" connection to the relay server is established" file=qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:277 category=qml
```

This issue also happens to me for:
- L1 Ethereum to Arbitrum L2
 https://github.com/user-attachments/assets/c83a159b-abf2-43e3-b33b-9fb57bed53ef

- Arbitrum L2 to L1 Ethereum
https://github.com/user-attachments/assets/0bf3739d-6501-4ce1-879e-1d4ccd730c42

However, I can bridge from L2 Arbitrum to L2 Optimism

In short, the only bridge path that works for me is L2 to L2. Am I missing any step to perform a successful bridge using these paths?

I'm attaching the console log and geth.log, hope this helps! 

[geth.log.zip](https://github.com/user-attachments/files/17260365/geth.log.zip)
[app_20241004_110649.log](https://github.com/user-attachments/files/17260367/app_20241004_110649.log)

**Comment by alexjba** [comment]
zapper doesn't even fetch my wallet balance. It's always 0 and I can't start any transaction.

**Comment by virginiabalducci** [comment]
Bridging from L2 to L2 works
Bridging from L1 to L2 gets stuck
![Image](https://github.com/user-attachments/assets/99f505d3-817e-48f6-972a-9c0826bbd350)

cc @alexjba

**Comment by virginiabalducci** [comment]
Bridging L2 -> L2 works
Swapping eth -> erc20 using Optimism and Arbitrum work. 
Swapping on Mainnet is too expensive to test at this moment (45usd fees)

Bridging L1 > L2 (Optimism or Arbitrum) does not work, it gets stuck on the Confirm page of the dApp as described in the previous comment. I'm attaching log files, I hope they are useful @alexjba 


[app_20241126_151312.log](https://github.com/user-attachments/files/17924352/app_20241126_151312.log)
[app_20241122_110054.log](https://github.com/user-attachments/files/17924353/app_20241122_110054.log)
[geth.log.zip](https://github.com/user-attachments/files/17924358/geth.log.zip)

**Comment by alexjba** [comment]
@virginiabalducci I'd close it as dApp issue.

The issue I was worried about was that we're not successful completing the tx received from the dApp, but it's unconfirmed. Looking on all the logs here I can see that all transactions are completed and we're forwarding to the dApp the correct tx hash.

I see two issues on dApp side:
1. Cannot initiate the tx: Probably different reasons - one we can see in the video is that the dApp cannot estimate Gas
2. Sometimes the dApp cannot validate the tx hash, although all the tx hases found in logs are correct.

**Comment by virginiabalducci** [comment]
I agree!
Closing this one as per the above comments.
